pkg/operator/staticpod/controller/node/node_controller.go

```diff
@@ -103,10 +101,11 @@ func (c NodeController) sync() error {
 		c.eventRecorder.Eventf("MasterNodeObserved", "Observed new master node %s", node.Name)
 		newTargetNodeStates = append(newTargetNodeStates, operatorv1.NodeStatus{NodeName: node.Name})
 	}
+
+	operatorStatus := originalOperatorStatus.DeepCopy()
 	operatorStatus.NodeStatuses = newTargetNodeStates
 
-	if !reflect.DeepEqual(originalOperatorStatus, operatorStatus) {
-		_, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus)
+	if _, updateError := c.operatorConfigClient.UpdateStatus(resourceVersion, operatorStatus); updateError != nil {
```

Comment:
you lost the deep equals check.  This will always update now.  It's late in europe now, I'll grab it, fix this, and merge